ironmon_tracker/Battle.lua

```diff
@@ -436,6 +473,43 @@ function Battle.beginNewBattle()
 		RightOwn = 2,
 		RightOther = 2,
 	}
+	--populate BattleAbilities for all Pokemon
+	Battle.BattleAbilities[0] = {}
+	Battle.BattleAbilities[1] = {}
+	for i=1, 6, 1 do
+		local ownPokemon = Tracker.getPokemon(i, true)
+		if ownPokemon ~= nil then
+			local ability = PokemonData.getAbilityId(ownPokemon.pokemonID, ownPokemon.abilityNum)
+			Battle.BattleAbilities[0][i] = {
+				abilityOwner = {
+					isOwn = true,
+					slot = i
+				},
+				["originalAbility"] = ability,
+				["ability"] = ability,
+				transformData = {
+					isOwn = true,
+					slot = i,
+				},
+			}
+		end
+		local enemyPokemon = Tracker.getPokemon(i, false)
+		if enemyPokemon ~= nil then
+			local ability = PokemonData.getAbilityId(enemyPokemon.pokemonID, enemyPokemon.abilityNum)
+			Battle.BattleAbilities[1][i] = {
+				abilityOwner = {
+					isOwn = false,
+					slot = i
+				},
+				["originalAbility"] = ability,
+				["ability"] = ability,
+				transformData = {
+					isOwn = false,
+					slot = i,
+				},
+			}
+		end
+	end
```

Comment:
Is this something can be all be put into a separate function, to help keep the "begin battle" function better organized and clean to read. I am also curious about the purpose behind what is going on here. I understand it reads populate battle abilities, but im not sure what that means or what thats for. Can chat about this later if you'd like

Reply:
Yes. BattleAbilites is the object which records changes to Abilities or identity (think Transform) over the course of a battle, this is just initializing it with both trainers' pokemon's starting details. So instead of logging whatever ability a pokemon has for them, the program will now know which pokemon that ability came from, and track it for the appropriate pokemon, even if they are no longer one of the 2/4 pokemon in the battle